code/modules/food_and_drinks/food/foods/baked_goods.dm

```diff
@@ -512,6 +870,232 @@
+	name = "jelly chocolate donut"
+	desc = "Goes great with a glass of warm milk."
+	icon_state = "jelly_choc"
+	extra_reagent = "slimejelly"
+
+/obj/item/food/donut/matcha
+	name = "matcha donut"
+	desc = "Goes great with a cup of tea."
+	icon_state = "donut_olive"
+	list_reagents = list("nutriment" = 3, "sugar" = 2, "teapowder" = 2)
+	filling_color = "#D2691E"
+	tastes = list("donut" = 1, "matcha" = 1)
+	randomized_sprinkles = FALSE
+
+/obj/item/food/donut/matcha/jelly
+	name = "jelly matcha donut"
+	desc = "Goes great with a cup of tea."
+	icon_state = "jelly_olive"
+	extra_reagent = "berryjuice"
+
+/obj/item/food/donut/matcha/jelly/cherryjelly
+	name = "jelly matcha donut"
+	desc = "Goes great with a cup of tea."
+	icon_state = "jelly_olive"
+	extra_reagent = "cherryjelly"
+
+/obj/item/food/donut/matcha/jelly/slimejelly
+	name = "jelly matcha donut"
+	desc = "Goes great with a cup of tea."
+	icon_state = "jelly_olive"
+	extra_reagent = "slimejelly"
+
+/obj/item/food/donut/bungo
+	name = "bungo donut"
+	desc = "Goes great with a mason jar of hippie's delight."
+	icon_state = "donut_yellow"
+	list_reagents = list("nutriment" = 3, "sugar" = 2, "bungojuice" = 3)
+	filling_color = "#D2691E"
+	tastes = list("donut" = 1, "tropical sweetness" = 1, "an acidic, poisonous tang" = 1)
+	randomized_sprinkles = FALSE
+
+/obj/item/food/donut/bungo/jelly
+	name = "jelly bungo donut"
+	desc = "Goes great with a mason jar of hippie's delight."
+	icon_state = "jelly_yellow"
+	extra_reagent = "berryjuice"
+
+/obj/item/food/donut/bungo/jelly/cherryjelly
+	name = "jelly bungo donut"
+	desc = "Goes great with a mason jar of hippie's delight."
+	icon_state = "jelly_yellow"
+	extra_reagent = "cherryjelly"
+
+/obj/item/food/donut/bungo/jelly/slimejelly
+	name = "jelly bungo donut"
+	desc = "Goes great with a mason jar of hippie's delight."
+	icon_state = "jelly_yellow"
+	extra_reagent = "slimejelly"
+
+/obj/item/food/donut/spaceman
+	name = "spaceman's donut"
+	desc = "Goes great with a cold beaker of malk."
+	icon_state = "donut_purple"
+	list_reagents = list("nutriment" = 3, "sugar" = 2)
+	filling_color = "#D2691E"
+	tastes = list("donut" = 1, "violets" = 1)
+	randomized_sprinkles = FALSE
+
+/obj/item/food/donut/spaceman/jelly
+	name = "jelly spaceman's donut"
+	desc = "Goes great with a cold beaker of malk."
+	icon_state = "jelly_purple"
+	extra_reagent = "berryjuice"
+
+/obj/item/food/donut/spaceman/jelly/cherryjelly
+	name = "jelly spaceman's donut"
+	desc = "Goes great with a cold beaker of malk."
+	icon_state = "jelly_purple"
+	extra_reagent = "cherryjelly"
+
+/obj/item/food/donut/spaceman/jelly/slimejelly
+	name = "jelly spaceman's donut"
+	desc = "Goes great with a cold beaker of malk."
+	icon_state = "jelly_purple"
+	extra_reagent = "slimejelly"
+
+/obj/item/food/donut/meat
+	name = "Meat Donut"
+	desc = "Tastes as gross as it looks."
+	icon_state = "donut_meat"
+	list_reagents = list("nutriment" = 3, "protein" = 3, "ketchup" = 3)
```

Comment:
This contains ketchup but the recipe doesn't require neither ketchup nor tomatoes